types/mongodb/test/collection/findX.ts

```diff
@@ -39,6 +39,15 @@ async function run() {
     sort: {stringField: -1, text: {$meta: 'textScore'}, notExistingField: -1}
   });
 
+  // 2nd element of sort sub-array can only be 1 or -1
+  await collectionT.findOne({}, {
+    projection: {
+      stringField: {$meta: 'textScore'},
+      fruitTags: {$min: 'fruitTags'},
+      max: {$max: ['$max', 0]},
+    },
+    sort: [['stringField', -1], ['numberField', 1]],
+  });
```

Comment:
I also tried to create the failing version of this test with:

```javascript
  await collectionT.findOne({}, {
    projection: {
      stringField: {$meta: 'textScore'},
      fruitTags: {$min: 'fruitTags'},
      max: {$max: ['$max', 0]},
    },
    sort: [['stringField', -5], ['numberField', 'i am a string']],
  });
```

But ran into:

```
No overload matches this call.

  Overload 1 of 3, '(filter: FilterQuery<TestModel>, callback: MongoCallback<TestModel | null>):
 void', gave the following error.

    Argument of type '{ projection: { stringField: { $meta: string; }; fruitTags: { $min: string; }; 
max: { $max: (string | number)[]; }; }; sort: (string | number)[][]; }' is not assignable to 
parameter of type 'MongoCallback<TestModel | null>'.

      Object literal may only specify known properties, and 'projection' does not exist in type 
'MongoCallback<TestModel | null>'.ts(2769)
```

I couldn't work out how to force tsc to use the specific overload that this would use. Any help here would be much appreciated.

Reply:
You should add `// $ExpectError` to specify some part of code should fail due to type mismatch.

Reply:
Agreed @HitkoDev - this test is the happy path - are you saying that we don't need a happy path test and only the failing one?

I wasn't able to work out how to get a failing test with $ExpectError as per the above tsc error because I wasn't able to force tsc to use a specific overload. Do you have any ideas around this?

Reply:
Both tests are necessary. For the failing case, try isolating the part you're tying to test; something like

```ts
const failingOptions: FindOneOptions<TestModel> =  {
    projection: {
      stringField: {$meta: 'textScore'},
      fruitTags: {$min: 'fruitTags'},
      max: {$max: ['$max', 0]},
    },
    // $ExpectError
    sort: [['stringField', -5], ['numberField', 'i am a string']],
  }
```

Reply:
Alternatively, you could try

```ts
const findOptions: FindOneOptions<TestModel> =  {
    projection: {
      stringField: {$meta: 'textScore'},
      fruitTags: {$min: 'fruitTags'},
      max: {$max: ['$max', 0]},
    },
    sort: [['stringField', -1], ['numberField', 1]],
  }
// $ExpectError
findOptions.sort = [['stringField', -5], ['numberField', 'i am a string']];
```

Reply:
Nice - I removed the method call completely for the tests and just used the FindOneOptions object in a happy/failed scenario. Thanks for those ideas @HitkoDev